tests/src/Kernel/SmartdocRoutingTest.php

```diff
@@ -61,37 +74,48 @@ class SmartdocRoutingTest extends KernelTestBase {
   protected function setUp() {
     parent::setUp();
 
-    $this->installEntitySchema('user');
-    $this->installEntitySchema('apidoc');
-
-    $this->apidoc = ApiDoc::create([
-      'name' => 'API 1',
-      'description' => 'Test API 1',
-      'spec' => NULL,
-      'api_product' => NULL,
-    ]);
-
-    $this->apidoc->save();
-
-    // Prepare to create a user.
-    $this->installEntitySchema('user');
     $this->installSchema('system', ['sequences']);
     $this->installSchema('user', ['users_data']);
+    $this->installSchema('node', ['node_access']);
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('node_type');
+    $this->installEntitySchema('path_alias');
+    $this->installConfig(static::$modules);
+
+    $this->apidoc = $this->container->get('entity_type.manager')
+      ->getStorage('node')
+      ->create([
+        'type' => 'apidoc',
+        'title' => 'API 1',
+        'body' => [
+          'value' => 'Test API 1',
+          'format' => 'basic_html',
+        ],
+        'field_apidoc_spec' => NULL,
+        'field_apidoc_api_product' => NULL,
+      ]);
 
-    // Rendering an apidoc requires the default filter formats be installed.
-    $this->installConfig(['filter']);
+    $this->apidoc->save();
 
-    $user = $this->createUser(['view published apidoc entities']);
+    $user = $this->createUser(['access content']);
     $this->setCurrentUser($user);
-
   }
 
   /**
    * Tests the route subscriber will redirect from smartdoc routes.
    */
   public function testNotFoundSubscriber() {
```

Comment:
There was 1 failure:
```
1) Drupal\Tests\apigee_api_catalog\Kernel\SmartdocRoutingTest::testNotFoundSubscriber
Failed asserting that 404 is identical to 302.

/Users/chrisnovak/projects/drupal8/drupal-api-catalog/web/modules/contrib/apigee_api_catalog/tests/src/Kernel/SmartdocRoutingTest.php:127
```

Reply:
This test was fixed to (converted to functional because path aliases were complicated in kernel tests).